mmseg/models/backbones/mscan.py

```diff
@@ -0,0 +1,526 @@
+                padding=(1, 1)),
+            build_norm_layer(norm_cfg, out_channels // 2)[1],
+            build_activation_layer(act_cfg),
+            nn.Conv2d(
+                out_channels // 2,
+                out_channels,
+                kernel_size=(3, 3),
+                stride=(2, 2),
+                padding=(1, 1)),
+            build_norm_layer(norm_cfg, out_channels)[1],
+        )
+
+    def forward(self, x):
+        """
+        Args:
+            x (Tensor): Has shape (B, C, H, W).
+                In most case, C is `in_channels`.
+
+        Returns:
+            tuple: Contains merged results and its spatial shape.
+                - x (Tensor): Has shape (B, out_h * out_w, embed_dims)
+                - H (int): Height of x.
+                - W (int): Width of x.
+        """
+        x = self.proj(x)
+        _, _, H, W = x.size()
+        x = x.flatten(2).transpose(1, 2)
+        return x, H, W
+
+
+class AttentionModule(BaseModule):
+    """Attention Module.
+
+    Args:
+        channels (int): The dimension of channels.
+        kernel_sizes (List[int | Tuple]): The size of attention
+            kernel. Defaults: [5, (1, 7), (1, 11), (1, 21)].
+        kernel_paddings (List[int | Tuple]): The number of
+            corresponding padding value in attention module.
+            Defaults: [2, (0, 3), (0, 5), (0, 10)].
+    """
+
+    def __init__(self,
+                 channels,
+                 kernel_sizes=[[5], [1, 7], [1, 11], [1, 21]],
+                 kernel_paddings=[2, (0, 3), (0, 5), (0, 10)]):
+        super().__init__()
+        for index_a, kernel_size in enumerate(kernel_sizes):
+            if index_a == 0:
+                conv_name = f'conv{index_a}'
+                self.add_module(
+                    conv_name,
+                    nn.Conv2d(
+                        channels,
+                        channels,
+                        kernel_sizes[index_a][index_a],
+                        padding=kernel_paddings[index_a],
+                        groups=channels))
+            else:
+                for index_b, k in enumerate(kernel_size):
+                    conv_name = f'conv{index_a-1}_{index_b+1}'
+                    if index_b == 0:
+                        kernel_size = (kernel_sizes[index_a][0],
+                                       kernel_sizes[index_a][1])
+                        padding = kernel_paddings[index_a]
+                    else:
+                        kernel_size = (kernel_sizes[index_a][1],
+                                       kernel_sizes[index_a][0])
+                        padding = kernel_paddings[index_a][::-1]
+                    self.add_module(
+                        conv_name,
+                        nn.Conv2d(
+                            channels,
+                            channels,
+                            kernel_size,
+                            padding=padding,
+                            groups=channels))
```

Comment:
what dose `k` in line 170 work for? I think this implementation needs to be refined.